ksqldb-engine/src/test/java/io/confluent/ksql/planner/plan/JoinNodeTest.java

```diff
@@ -638,10 +638,14 @@ public void shouldHaveFullyQualifiedJoinSchemaWithNonSyntheticKey() {
         .valueColumn(ColumnName.of(LEFT_ALIAS.text() + "_C0"), SqlTypes.BIGINT)
         .valueColumn(ColumnName.of(LEFT_ALIAS.text() + "_L1"), SqlTypes.STRING)
         .valueColumn(ColumnName.of(LEFT_ALIAS.text() + "_ROWTIME"), SqlTypes.BIGINT)
+        .valueColumn(ColumnName.of(LEFT_ALIAS.text() + "_ROWPARTITION"), SqlTypes.INTEGER)
```

Comment:
How come these unit tests are being updated? I thought the feature flag was disabled by default?

Reply:
This is because `LogicalSchema.withPseudoAndKeyColsInValue(boolean)`, which uses the `CURRENT_PSEUDOCOLUMN_VERSION_NUMBER` is being used by some unit tests . Since for all of those tests (except this one) the results are unaffected by which version we are using, I thought it'd be easier for whoever ends up removing this flag to just leave the current constructor as an overload instead of fully replacing it with a `(boolean, KsqlConfig`) constructor. In fact, in it's current state, `LogicalSchema.withPseudoAndKeyColsInValue(boolean)` should be annotated with `@VisibleForTesting`.

It's definitely not a major issue to mock the `KsqlConfig` into each unit test that uses it if you think that'd make things more explicit. Or I could do it just for this unit test. Or I could just annotate the constructor. I lean towards the second, just mocking this unit test (not entirely sure why I didn't before). WDYT?

Reply:
As long as we have an understanding of why this is happening, and that understanding lends confidence to the fact that this PR is non-functional changes only (all the new functionality is hidden behind the feature flag), then we're good. I do think it's preferable to mock the KsqlConfig in this unit test file, though, similar to what you've done for the other unit tests.

Reply:
SGTM, I'll do that.